test/testinfra/MUXTestInfra/Common/InputHelper.cs

```diff
@@ -295,6 +309,74 @@ public static void TapAndHold(UIObject obj, uint durationMs)
             Wait.ForIdle();
         }
 
+        public static void DoubleTap(UIObject obj)
+        {
+            // Allowing three attempts to work around occasional failure on Phone builds.
+            int retries = 3;
+            while (retries > 0)
+            {
+                try
+                {
+                    Log.Comment("Double-tap on {0}.", obj.GetIdentifier());
+                    using (var waiter = GetWaiterForInputEvent(obj, InputEvent.Tap))
+                    {
+                        if (PlatformConfiguration.IsOSVersionLessThan(OSVersion.Redstone5))
+                        {
+                            Log.Warning("Touch input is not available on OS versions less than RS5. Falling back to mouse input.");
+                            obj.DoubleClick(PointerButtons.Primary);
+                        }
+                        else
+                        {
+                            obj.DoubleTap();
+                        }
+                    }
+
+                    retries = 0;
+                }
+                catch (Exception e)
+                {
+                    Log.Warning("Exception while double-tapping: " + e.Message);
```

Comment:
nit: A more accurate message would be
Log.Warning((PlatformConfiguration.IsOSVersionLessThan(OSVersion.Redstone5) ? "Exception while double-clicking: " : "Exception while double-tapping: ") + e.Message);

Reply:
There's already a warning message informing the developer that we fell back to mouse input, so I think I'd prefer to leave this as-is.